### PR TITLE
[Remove Blocking usages]LPS-186857 document-library-service | DLSubscriptionSender

### PR DESCRIPTION
**Pull request by W2jx**
resend

Previous: https://github.com/liferay-core-infra/liferay-portal/pull/4820

### CODE REVIEW

**Comment by W2jx** [comment]
ci:test:startup

**Comment by W2jx** [comment]
ci:test:document

**Comment by Kvale1733** [comment]
@W2jx No regressions found.

**Comment by W2jx** [comment]
no conflicts(6/27 8:52)

File: modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/util/DLSubscriptionSender.java [hunk 1]
```diff
@@ -55,12 +55,11 @@ protected Boolean hasSubscribePermission(
 		return super.hasSubscribePermission(permissionChecker, subscription);
 	}
 
-	private static volatile ModelResourcePermission<DLFolder>
-		_dlFolderModelResourcePermission =
-			ServiceProxyFactory.newServiceTrackedInstance(
-				ModelResourcePermission.class, DLSubscriptionSender.class,
-				"_dlFolderModelResourcePermission",
-				"(model.class.name=" + DLFolder.class.getName() + ")", true);
+	private static final Snapshot<ModelResourcePermission<DLFolder>>
```
Comment (dantewang) [inline modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/util/DLSubscriptionSender.java:58]: @W2jx I'm thinking about this, could you double check how this `DLSubscriptionSender` is used in the whole subscription sending process (not just the class that creates and calls method on it)? When will it, as a `SubscriptionSender`, be serialized and deserialized?

Comment (W2jx) [reply modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/util/DLSubscriptionSender.java:58]: Hi @dantewang , I followed the steps in this [ticket](https://liferay.atlassian.net/browse/LPS-157875) to test.

**steps：**

1. Enable Publications
2. Add a new publication
3. Navigate to DM and upload a png file
4. Checkout the file
5. Checkin the file with major version
6. Publish the publication

**test result：**

_1、Serialization_

When I execute step 3, class `DLSubscriptionSender` will be set as payload of the message and then the message will be serialized into json and stored in table `CTMessage`

Screenshot_2023-07-04_10-35-24


_2、deserialization_


When I execute step 6, the message stored in table  `CTMessage` will be got and deserialized into object.

Screenshot_2023-07-04_11-12-55

Comment (dantewang) [reply modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/util/DLSubscriptionSender.java:58]: @W2jx Is this subscription sender used after being deserialized?

**Comment by dantewang** [comment]
ci:test:stable

File: modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/util/DLSubscriptionSender.java (hunk 1 above)
Comment (W2jx) [reply modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/util/DLSubscriptionSender.java:58]: Hi @dantewang , yes, `DLSubscriptionSender` will be used in method ` SubscriptionSenderMessageListener.doReceive` after being deserialized.

The code：

https://github.com/liferay/liferay-portal/blob/330307173a79ce480d77696c4a7411d45c95bb13/modules/apps/subscription/subscription-service/src/main/java/com/liferay/subscription/internal/messaging/SubscriptionSenderMessageListener.java#L41-L56

Comment (dantewang) [reply modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/util/DLSubscriptionSender.java:58]: @W2jx I see, thanks! So it's used in a common place which is not aware of whether this is a `DLSubscriptionSender`.

**Comment by dantewang** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/137298

**Pull request closed without merging**